target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-dlink-dap-2610.dts

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-ap.dk01.1.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "D-Link DAP 2610";
+	compatible = "dlink,dap-2610";
+
+	soc {
+		edma@c080000 {
+			qcom,num_gmac = <1>;
+		};
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+		led-upgrade = &power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		red {
+			label = "dap-2610:red:power";
+			gpios = <&tlmm 4 GPIO_ACTIVE_LOW>;
+		};
+
+		power: green {
+			label = "dap-2610:green:power";
+			gpios = <&tlmm 5 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 63 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gmac0 {
+	vlan_tag = <2 0x20>;
+};
+
+&blsp1_spi1 {
+	flash@0 {
+		compatible = "w25q128", "jedec,spi-nor";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0>;
+		spi-max-frequency = <33000000>;
```

Comment:
Where is this from?
Other ipq40xx boards are running at 24000000

Reply:
It is from the w25q128 specification. I guess it is safer to user keep 24000000 here as well if D-Link decides to change chip or whatnot.

Reply:
Well, it cant runt at 33 MHz as IPQ4019 has a controller with max of 25 MHz.
That is why QSDK and everybody just uses 24 MHz to be safe.

Reply:
@robimarko A bit off-topic, but could you provide a reference to it? Without knowing it I made a commit 80839694056d raising frequency to 30MHz. If it's true, I guess we'd better to revert the commit.

Reply:
Actually, according to the driver comments and defines it appers that QUP controller can go up to 50MHz, anything above 26MHz is defined as high speed in the driver.

Reply:
Oh, so you were talking about this: https://elixir.bootlin.com/linux/latest/source/drivers/spi/spi-qup.c#L119
And that explains why ens620ext used 50MHz... https://github.com/openwrt/openwrt/blob/master/target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-ens620ext.dts#L167
So frequency above 24MHz is not totally prohibited... Thanks for the information. :)